inference-engine/tests/functional/inference_engine/ir_serialization/ordering.cpp

```diff
@@ -0,0 +1,165 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <fstream>
+
+#include "common_test_utils/ngraph_test_utils.hpp"
+#include "gtest/gtest.h"
+#include "ie_core.hpp"
+#include "pugixml.hpp"
+
+#ifndef IR_SERIALIZATION_MODELS_PATH  // should be already defined by cmake
+#define IR_SERIALIZATION_MODELS_PATH ""
+#endif
+
+namespace {
+// walker traverse (DFS) xml document and store layer type attriubte in
+// vector which is later used for comparison
+struct type_attribute_walker : pugi::xml_tree_walker {
+    std::vector<pugi::xml_attribute> attributes;
+
+    virtual bool for_each(pugi::xml_node& node) {
+        const std::string node_name{node.name()};
+        if (node_name == "layer") {
+            for (const auto& attribute : node.attributes()) {
```

Comment:
Suggested alternative:
```
std::copy_if(node.attributes().begin(), node.attributes().end(), std::back_inserter(attributes),
    [](const pugi::xml_attribute& attribute) {
        return std::string{attribute.name()} == "type";
    });
```

Reply:
Applied in [this](https://github.com/openvinotoolkit/openvino/pull/3156/commits/1b80d2856153c7ccad62f8cd9146934a34529b23) commit.